okta/requestExecutor.go

```diff
@@ -120,24 +120,36 @@ func (re *RequestExecutor) NewRequest(method string, url string, body interface{
 			token := re.cache.GetString(AccessTokenCacheKey)
 			req.Header.Add("Authorization", "Bearer "+token)
 		} else {
-			priv := []byte(strings.ReplaceAll(re.config.Okta.Client.PrivateKey, `\n`, "\n"))
 
-			privPem, _ := pem.Decode(priv)
-			if privPem == nil {
-				return nil, errors.New("invalid private key")
-			}
-			if privPem.Type != "RSA PRIVATE KEY" {
-				return nil, fmt.Errorf("RSA private key is of the wrong type")
-			}
+			var signer jose.Signer
+			switch key := re.config.Okta.Client.PrivateKey.(type) {
 
-			parsedKey, err := x509.ParsePKCS1PrivateKey(privPem.Bytes)
-			if err != nil {
-				return nil, err
-			}
+			case string:
+				priv := []byte(strings.ReplaceAll(key, `\n`, "\n"))
+
+				privPem, _ := pem.Decode(priv)
```

Comment:
Should this error here be used?

Reply:
This code was part of the original private key parsing flow. I'm not sure what the original context was for this decision as the error does appear to be flattened in the next line (if `privPem` is a `nil` value). In an ideal world, the caught error would be wrapped with `fmt.Errorf` before being returned.